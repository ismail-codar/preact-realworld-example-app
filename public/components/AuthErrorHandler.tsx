import { useStore } from '../store';

export function AuthErrorHandler() {
	const { error } = useStore(state => ({
		error: state.error
	}));

	return Object.keys(error).length === 0 ? null : (
		<ul class="error-messages">
			{Object.keys(error).map(key => (
				<li key={key} aria-label={`${key} error`}>
					{key} {error[key]}
				</li>
			))}
		</ul>
	);
}
