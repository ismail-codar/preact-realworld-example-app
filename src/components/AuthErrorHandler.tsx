import { h } from 'preact';

import useStore from '../store';

export default function AuthErrorHandler() {
	const error = useStore(state => state.error);

	return !error ? null : (
		<ul class="error-messages">
			{Object.keys(error).map(key => (
				<li key={key}>
					{key} {error[key]}
				</li>
			))}
		</ul>
	);
}
