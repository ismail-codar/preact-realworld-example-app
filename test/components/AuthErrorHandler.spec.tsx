import { h } from 'preact';
import { render, screen } from '@testing-library/preact';

import AuthErrorHandler from '../../public/components/AuthErrorHandler';
import useStore from '../../public/store';

jest.mock('../../public/store', () => jest.fn());

test('should return null when there are no errors', () => {
	((useStore as unknown) as jest.Mock).mockReturnValue({ error: {} });

	const { asFragment } = render(<AuthErrorHandler />);

	expect(asFragment()).toMatchInlineSnapshot(`<DocumentFragment />`);
});

test('should display auth errors when they exist', () => {
	((useStore as unknown) as jest.Mock).mockReturnValue({
		error: {
			username: ['has already been taken'],
			email: ['has already been taken'],
			password: ['is too short (minimum 8 characters)']
		}
	});
	const { asFragment } = render(<AuthErrorHandler />);

	expect(screen.getByRole('listitem', { name: 'username error' })).toHaveTextContent('username has already been taken');
	expect(screen.getByRole('listitem', { name: 'email error' })).toHaveTextContent('email has already been taken');
	expect(screen.getByRole('listitem', { name: 'password error' })).toHaveTextContent(
		'password is too short (minimum 8 characters)'
	);

	expect(asFragment()).toMatchSnapshot();
});
