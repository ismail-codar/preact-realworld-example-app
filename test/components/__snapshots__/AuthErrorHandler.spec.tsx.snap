// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display auth errors when they exist 1`] = `
<DocumentFragment>
  <ul
    class="error-messages"
  >
    <li
      aria-label="username error"
    >
      username has already been taken
    </li>
    <li
      aria-label="email error"
    >
      email has already been taken
    </li>
    <li
      aria-label="password error"
    >
      password is too short (minimum 8 characters)
    </li>
  </ul>
</DocumentFragment>
`;
